src/model/transaction/Transaction.ts

```diff
@@ -81,29 +88,54 @@ export abstract class Transaction {
 
     /**
      * Generate transaction hash hex
+     *
+     * @see https://github.com/nemtech/catapult-server/blob/master/src/catapult/model/EntityHasher.cpp#L32
+     * @see https://github.com/nemtech/catapult-server/blob/master/src/catapult/model/EntityHasher.cpp#L35
+     * @see https://github.com/nemtech/catapult-server/blob/master/sdk/src/extensions/TransactionExtensions.cpp#L46
      * @param {string} transactionPayload HexString Payload
      * @param {Array<number>} generationHashBuffer Network generation hash byte
      * @param {NetworkType} networkType Catapult network identifier
      * @returns {string} Returns Transaction Payload hash
      */
     public static createTransactionHash(transactionPayload: string, generationHashBuffer: number[], networkType: NetworkType): string {
-        const type = parseInt(Convert.uint8ToHex(Convert.hexToUint8(transactionPayload.substring(220, 224)).reverse()), 16);
-        const byteBuffer = Array.from(Convert.hexToUint8(transactionPayload));
-        const byteBufferWithoutHeader = byteBuffer.slice(4 + 64 + 32 + 8);
-        const dataBytes = type === TransactionType.AGGREGATE_BONDED || type === TransactionType.AGGREGATE_COMPLETE ?
-            generationHashBuffer.concat(byteBufferWithoutHeader.slice(0, 52)) :
-            generationHashBuffer.concat(byteBufferWithoutHeader);
-        const signingBytes = byteBuffer
-            .slice(8, 40) // first half of signature
-            .concat(byteBuffer
-                .slice(4 + 4 + 64, 8 + 64 + 32)) // signer
-            .concat(dataBytes);
 
-        const hash = new Uint8Array(32);
-        const signSchema = SHA3Hasher.resolveSignSchema(networkType);
-        SHA3Hasher.func(hash, signingBytes, 32, signSchema);
+        // prepare
+        const entityHash: Uint8Array = new Uint8Array(32);
+        const signSchema: SignSchema = SHA3Hasher.resolveSignSchema(networkType);
+        const transactionBytes: Uint8Array = Convert.hexToUint8(transactionPayload);
+
+        // 1) take "R" part of a signature (first 32 bytes)
+        const signatureR: Uint8Array = transactionBytes.slice(8, 8 + 32);
+
+        // 2) add public key to match sign/verify behavior (32 bytes)
+        const pubKeyIdx: number = signatureR.length;
+        const publicKey: Uint8Array = transactionBytes.slice(8 + 64, 8 + 64 + 32);
+
+        // 3) add generationHash (32 bytes)
+        const generationHashIdx: number = pubKeyIdx + publicKey.length;
+        const generationHash: Uint8Array = Uint8Array.from(generationHashBuffer);
+
+        // 4) add transaction data without header (EntityDataBuffer)
+        // @link https://github.com/nemtech/catapult-server/blob/master/src/catapult/model/EntityHasher.cpp#L30
+        const transactionBodyIdx: number = generationHashIdx + generationHash.length;
+        const transactionBody: Uint8Array = transactionBytes.slice(Transaction.Header_Size);
```

Comment:
Not sure this will work on the AggregateTransaction specifically. As the new `signingBytes` for Aggregated changed to only include the `merkleTransactionHash` field but not rest of the payload

Reply:
changed will be applied with next push